esmvaltool/cmorizers/data/downloaders/datasets/cds_satellite_soil_moisture.py

```diff
@@ -0,0 +1,81 @@
+"""Script to download CDS-SATELLITE-SOIL-MOISTURE from the CDS."""
+
+import calendar
+import datetime
+
+from dateutil import relativedelta
+
+from esmvaltool.cmorizers.data.downloaders.cds import CDSDownloader
+from esmvaltool.cmorizers.data.utilities import unpack_files_in_folder
+
+
+def download_dataset(config, dataset, dataset_info, start_date, end_date,
+                     overwrite):
+    """Download dataset.
+
+    Parameters
+    ----------
+    config : dict
+        ESMValTool's user configuration
+    dataset : str
+        Name of the dataset
+    start_date : datetime
+        Start of the interval to download
+    end_date : datetime
+        End of the interval to download
+    overwrite : bool
+        Overwrite already downloaded files
+    """
+    if not start_date:
+        start_date = datetime.datetime(1978, 11, 1)
+    if not end_date:
+        end_date = datetime.datetime(2020, 6, 30)
+
+    loop_date = start_date
+    downloader = CDSDownloader(
+        product_name='satellite-soil-moisture',
+        request_dictionary={
+            'format': 'tgz',
+            'variable': 'volumetric_surface_soil_moisture',
+            'type_of_sensor': 'combined_passive_and_active',
+            'type_of_record': 'cdr',
+            'version': 'v201912.0.0',
+            'time_aggregation': 'month_average',
+            'day': ['01']
+        },
+        config=config,
+        dataset=dataset,
+        dataset_info=dataset_info,
+        overwrite=overwrite,
+    )
+
+    while loop_date <= end_date:
+        downloader.download(loop_date.year, loop_date.month)
+        loop_date += relativedelta.relativedelta(months=1)
+
+    downloader = CDSDownloader(
+        product_name='satellite-soil-moisture',
+        request_dictionary={
+            'format': 'tgz',
+            'variable': 'volumetric_surface_soil_moisture',
+            'type_of_sensor': 'combined_passive_and_active',
+            'day': '01',
+            'type_of_record': 'cdr',
+            'version': 'v201912.0.0',
+            'time_aggregation': 'day_average',
+        },
```

Comment:
In #1851, support was added for different configurations so we now have 2 time aggregation options (daily and monthly) and 3 type of censor options (active, passive, combined_passive_and_active), totalling 6 possibilities. Do you think all 6 should be included in this script? If so, several request_dictionary may be needed. For example, for monthly averages, we would have two more cases than the one already used:
```python
            'format': 'tgz',
            'variable': 'volumetric_surface_soil_moisture',
            'type_of_sensor': 'passive',
            'type_of_record': 'cdr',
            'version': 'v201912.0.0',
            'time_aggregation': 'month_average',
            'day': ['01']
```
and
```python
            'format': 'tgz',
            'variable': 'surface_soil_moisture',
            'type_of_sensor': 'active',
            'type_of_record': 'cdr',
            'version': 'v201912.0.0',
            'time_aggregation': 'month_average',
            'day': ['01']
```
I'm not sure if and how all these cases should be included.

Reply:
I changed the default to get all 6 cases. probably the next step in this is to allow users to select specific variables and versions without modifying any files, but it is out of the scope of this PR

Reply:
Perfect! That works fine for me as well. Agreed, we should open an issue to keep that in mind once this is merge. This would be a nice improvement in the future.